capstone-frontend/src/components/CreateList.jsx

```diff
@@ -64,28 +126,86 @@ class CreateList extends Component {
           </Modal.Header>
           <Modal.Body>
             <Form>
-              <Form.Group controlId="createBookListForm">
+              <Form.Group controlId="form-booklist-name">
                 <Form.Label>Name of Booklist</Form.Label>
                 <Form.Control type="text" placeholder="Enter Name" />
               </Form.Group>
-              <Button variant="primary" type="submit" onClick={() => this.handleSubmit()}disabled={this.state.loading}>
-                Create Booklist
-                    {this.state.loading &&
-                  <Spinner
-                    as="span"
-                    animation="border"
-                    size="sm"
-                    role="status"
-                    aria-hidden="true"
-                    className="ml-4"
-                  />}
-              </Button>
+              <Form.Group controlId="form-search-term">
+                <Form.Label>Search for Books</Form.Label>
+                <Form.Control type="text" placeholder="Search for books to add" onChange={(event) => this.handleSearchTermChange(event)} />
+                {this.state.fetchingBooks &&
+                  <div className="text-center">
+                    <Spinner
+                      as="span"
+                      animation="border"
+                      size="lg"
+                      role="status"
+                      aria-hidden="true"
+                      className="my-5"
+                    />
+                  </div>}
+              </Form.Group>
+
+              {
+                this.state.displayBooks &&
```

Comment:
Being able to perform searches in the modal and add to lists here is awesome! This functionality would also be super useful in updating existing lists, and updating the book a club is reading.

Reply:
I agree, this is really cool!